### DISCUSSION TITLE
$chatMember->status , it seems not defined

### DISCUSSION THREAD

**Discussion by ghost** [nutgram/nutgram#370] [Q&A] [answered]
![Screenshot from 2023-03-05 12-11-08](https://user-images.githubusercontent.com/78247242/222950612-1cc6e4fd-5a11-4994-9159-e673041b6194.png)

```json
{
    "ok": true,
    "result": {
        "user": {
            "id": 1816677960,
            "is_bot": false,
            "first_name": "pooria",
            "language_code": "en"
        },
        "status": "creator",
        "is_anonymous": false
    }
}
```
The same is true for ‍`is_anonymous` ( `$chatMember->is_anonymous` )
What is the problem? ( Nutgram version: 3.15.2 )

**Comment by Lukasss93** [comment, marked as answer]
The IDE will not autocomplete that fields because `ChatMember` is not a [real type](https://core.telegram.org/bots/api#chatmember) (it's an abstract type)...

![immagine](https://user-images.githubusercontent.com/4071613/222968715-066705ba-db4e-46f5-a1dd-f8a49311bfba.png)

Anyway, to fix this, you can check the specific type of `ChatMember`, like this:

```php
$chatMember = $bot->getChatMember($chatID, $userID);

if($chatMember instanceof ChatMemberMember || 
   $chatMember instanceof ChatMemberAdministrator || 
   $chatMember instanceof ChatMemberOwner) {
    // your code
}
```

**Reply by ghost** [reply]
Thank you bro.
What about this ?! :

![Screenshot from 2023-03-06 00-04-17](https://user-images.githubusercontent.com/78247242/222984459-53495695-52fa-41d7-9747-ff92e986d413.png)



In old versions of Nutgram, these issues did not exist. These are a bit strange... I'm a bit confused.


editor error:

```
@var null $repliedMessage
Expected type 'object'. Found 'null'.intelephense(1006)
```

**Reply by ghost** [reply]
Ahh shit. it's working (the second issue). the problem is the if statement `if ($repliedMessage)` : /

Anyway, Thank you again :)
